agent/api/task/service_connect_validation.go

```diff
@@ -0,0 +1,293 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package task
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+// validateContainerName validates the service connect container name.
+func validateContainerName(scContainerName string, taskContainers []*ecsacs.Container) error {
+	// service connect container name is a required field
+	if scContainerName == "" {
+		return fmt.Errorf("missing service connect container name")
+	}
+
+	// validate the specified service connect container name exists in the task definition
+	numOfFoundSCContainer := 0
+	for _, container := range taskContainers {
+		if aws.StringValue(container.Name) == scContainerName {
+			numOfFoundSCContainer += 1
+		}
+	}
+
+	if numOfFoundSCContainer == 0 {
+		return fmt.Errorf("service connect container name=%s does not exist in the task", scContainerName)
+	} else if numOfFoundSCContainer > 1 {
+		return fmt.Errorf("found %d duplicate service connect container name=%s exists in the task", numOfFoundSCContainer, scContainerName)
+	}
+
+	return nil
+}
+
+// validateEgressConfig validates the service connect egress config.
+func validateEgressConfig(scEgressConfig *EgressConfig, ipv6Enabled bool) error {
+	// egress config can be empty for the first service since there are no other tasks that it can talk to
+	if scEgressConfig == nil {
+		return nil
+	}
+
+	// ListenerName is a required field if the egress config is not empty
+	if scEgressConfig.ListenerName == "" {
+		return fmt.Errorf("no service connect listener name")
```

Comment:
nit: given we have both ingress and egress listener, could we be specific here in the error messsage?